internal/iamrolesanywhere/credentials_test.go

```diff
@@ -0,0 +1,110 @@
+package iamrolesanywhere_test
+
+import (
+	"context"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	"go.uber.org/zap"
+
+	"github.com/aws/eks-hybrid/internal/api"
+	"github.com/aws/eks-hybrid/internal/iamrolesanywhere"
+)
+
+func TestWaitForAWSConfig(t *testing.T) {
+	tests := []struct {
+		name                  string
+		enableCredentialsFile bool
+		createCredsFile       bool
+		credsFileDelay        time.Duration
+		ctxTimeout            time.Duration
+		expectedError         string
+		expectedAccessKey     string
+		expectedSecretKey     string
+	}{
+		{
+			name:                  "success with credentials file",
+			enableCredentialsFile: true,
+			createCredsFile:       true,
+			credsFileDelay:        2 * time.Millisecond,
+			ctxTimeout:            5 * time.Second,
```

Comment:
nit: can we make this faster? like 10 millis or something
these longer tests add up when you want to run all of them
And i dont think waiting for longer provides more coverage